src/rgw/rgw_op.cc

```diff
@@ -5284,6 +5284,62 @@ int RGWCopyObj::verify_permission(optional_yield y)
 					      RGW_PERM_READ)) { 
 	  return -EACCES;
 	}
+
+  // Check for tagging permissions
+  auto tagging_drctv = s->info.env->get("HTTP_X_AMZ_TAGGING_DIRECTIVE");
+  TaggingDirective itag_der = TaggingDirective::NO_TAGGING_DIRECTIVE;
+  if (tagging_drctv) {
+    if (strcmp(tagging_drctv, "COPY") == 0)
+      itag_der = TaggingDirective::COPY_TAGGING_DIRECTIVE;
+    else if (strcmp(tagging_drctv, "REPLACE") == 0)
+      itag_der = TaggingDirective::REPLACE_TAGGING_DIRECTIVE;
+    else{
+      ldpp_dout(this, 0) << "ERROR: Invalid tagging directive " << dendl;
+      s->err.message = "Unknown tagging directive.";
+      return -EINVAL;
+    }
+  } else {
```

Comment:
'else' case not needed, at Line 5293 itag_der is by default initialized with NO_TAGGING_DIRECTIVE

Reply:
Done